### ISSUE TITLE
BaseWrapper doesn't play nice with fix.factors.prediction

### ISSUE THREAD

**Issue by mb706** [mlr-org/mlr#2411]
**tl;dr:** `predictLearner.BaseWrapper` should probably call `predictLearner2` instead of `predictLearner`, because the `fix.factors.prediction` prediction-side step is otherwise skipped for wrapped learners. Alternative solution would be to put the `fix.factors.prediction` logic into the `predictLearner` (generic) function

---

When wrapping (e.g. using a CPO or `makeXXXWrapper`) a Learner with `fix.factors.prediction` set to `TRUE`, the `train()` call chain goes along `train[wrapper]()` -> `train[base model]()`; each of the `train` calls may modify the training data according to the specific Learner's `fix.factors.prediction` and save the `factor.levels` in the resulting `WrappedModel` (which is then chained using `makeChainModel`).

During the `predict()` call, however, the call chain is `predict()` -> `predictLearner2()` -> `predictLearner[wrapper]()` -> `predictLearner[base model]()`. The `fix.factors.prediction` part happens in `predictLearner2()`, however, so it is skipped for the inner model.

Superficially one may think that the wrapper function (`makeXXXWrapper`) should simply *inherit* the `fix.factors.prediction` value from the base Learner (**\***); this would break, however, in cases where the wrapping Learner (e.g. a CPO or Impute Wrapper) changes task factors.

Example code: The first two invocations of `resample` go fine (setting `wlrn$fix.factors.prediction = TRUE` to demonstrate that the `classif.randomForest` does not fail because of `fix.factors.prediction` alone!), while the last invocation fails with a `New factor levels not present in the training data` error.
```R
test <- makeClassifTask("test",
  data.frame(a = factor(c("a", "b", "a", NA)),
    b = factor(c("a", "b", "a", "b"))),
  target = "b")

rflrn <- makeLearner("classif.randomForest")
wlrn <- makeImputeWrapper(rflrn,
  classes = list(factor = imputeConstant("MISSING")))
resample(wlrn, test, makeResampleDesc("LOO"))
# fine

rflrn <- makeLearner("classif.randomForest")
wlrn <- makeImputeWrapper(rflrn,
  classes = list(factor = imputeConstant("MISSING")))
wlrn$fix.factors.prediction = TRUE
resample(wlrn, test, makeResampleDesc("LOO"))
# fine

rflrn <- makeLearner("classif.randomForest")
rflrn$fix.factors.prediction = TRUE
wlrn <- makeImputeWrapper(rflrn,
  classes = list(factor = imputeConstant("MISSING")))
resample(wlrn, test, makeResampleDesc("LOO"))
# Error in predict.randomForest(.model$learner.model, newdata = .newdata,  : 
#   New factor levels not present in the training data
```

---

\* **Bonus Bug**: This might be why `makeModelMultiplexer` sets `fix.factors.prediction` generally to `TRUE`? When this is fixed, `makeModelMultiplexer` should not do that any more; instead, each base Learner should individually control whether it fixes prediction factors.
